pkg/service/pipelinerun.go

```diff
@@ -140,7 +141,10 @@ func (s *service) ListPipelineRuns(ctx context.Context, req *pb.ListPipelineRuns
 		if err != nil {
 			return nil, fmt.Errorf("failed to convert pipeline run: %w", err)
 		}
-		pbRun.RunnerId = runnerMap[run.TriggeredBy]
+		pbRun.RunnerId = userIDMap[run.TriggeredBy]
+		if requesterID, ok := userIDMap[run.Namespace]; ok && requesterID != nil {
```

Comment:
nit: Ideally I'd use the same strategy for both fields (empty string instead of `null` if the value doesn't exist). If you must use both approaches, I'd favour handling strings instead of pointers. Even if the code is simple, with the former you never risk accessing a `null` pointer.

Reply:
That is reasonable but I don't think it should be included in this ticket. We could update the RunnerId to string instead of string pointer later.